src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'blue' | 'green' | 'red';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle dimensions');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid side lengths');
```

Comment:
[CODE STYLE] - The error message 'Invalid side lengths' should be more descriptive. For example, 'Triangle side lengths must be greater than zero.'